File: 3264/main.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(getFinalState([]int{2,1,3,5,6}, 5, 2))
	fmt.Println(getFinalState([]int{7,71,15}, 10, 3))	// [567, 639, 1215]
}

func getFinalState(nums []int, k int, multipliter int) []int {
	for i := 0; i < k; i++ {
		idx := get_min(nums)
		nums[idx] = nums[idx] * multipliter
	}
	return nums
}

func get_min(nums []int) int {
	m := 100000000
	idx := 0
	for i := range nums {
		if nums[i] < m {
			m = nums[i]
			idx = i
		}
	}
	return idx
}

File: 3264/main.rs
struct Solution {}

impl Solution {
    pub fn get_final_state(mut nums: Vec<i32>, k: i32, multiplier: i32) -> Vec<i32> {
        for _ in 0..k {
            let mut m = 0;
            for j in 1..nums.len() {
                if nums[j] < nums[m] {
                    m = j;
                }
            }
            nums[m] *= multiplier;
        }
        nums
    }
}

fn main() {
    println!("{:#?}", Solution::get_final_state(vec![1,2], 3, 4));
}